### PR TITLE
Resolved `error.error` error

### LINKED ISSUE #1707: error.error is undefined when canceling a download
While debugging another issue, I came across this error:
> Uncaught (in promise) TypeError: error.error is undefined
> message: Object { dispatchConfig: null, _targetInst: null, _dispatchListeners: null, … }
in FilesContext.tsx:665:54

How to reproduce:
- try to download a single file
- cancel the download

### PR DESCRIPTION
**Pull request by RyRy79261**
closes #1707

### CODE REVIEW

File: packages/files-ui/src/Contexts/FilesContext.tsx [hunk 1]
```diff
@@ -662,7 +662,10 @@ const FilesProvider = ({ children }: FilesContextProps) => {
       return Promise.resolve()
     } catch (error: any) {
       console.error(error)
-      let errorMessage = `${t`An error occurred: `} ${typeof(error) === "string" ? error : error.error.message ? error.error.message : ""}`
+      let errorMessage = `${t`An error occurred: `} ${typeof(error) === "string"
+        ? error : error.message
+          ? error.message : error.error.message
+            ? error.error.message : ""}`
```
Comment (Tbaut) [inline packages/files-ui/src/Contexts/FilesContext.tsx]: Would something like this work?
```suggestion
      let errorMessage = `${t`An error occurred: `} ${error || error.message || error.error.message || ""}
```

This suggestion still looks awful to my eyes, but a little bit better than 3 ternaries.

Comment (RyRy79261) [reply packages/files-ui/src/Contexts/FilesContext.tsx]: Yeah was just following the existing approach, this is much nicer

File: packages/files-ui/src/Contexts/FilesContext.tsx [hunk 2]
```diff
@@ -662,7 +662,7 @@ const FilesProvider = ({ children }: FilesContextProps) => {
       return Promise.resolve()
     } catch (error: any) {
       console.error(error)
-      let errorMessage = `${t`An error occurred: `} ${typeof(error) === "string" ? error : error.error.message ? error.error.message : ""}`
+      let errorMessage = `${t`An error occurred: `} ${error || error.message || error.error.message || ""}
```
Comment (Tbaut) [inline packages/files-ui/src/Contexts/FilesContext.tsx]: previous suggestion gone wrong. BTW this should be tested, I did not test it :D
```suggestion
      let errorMessage = `${t`An error occurred: `} ${error || error.message || error.error.message || ""}`
```

Comment (tanmoyAtb) [reply packages/files-ui/src/Contexts/FilesContext.tsx]: Wouldn't this way create problems if `error` was an object and not a string ?

Comment (FSM1) [reply packages/files-ui/src/Contexts/FilesContext.tsx]: My preference here would be to just use 
```suggestion
      let errorMessage = `${t`An error occurred: `} ${typeof(error) === "string" ? error : error?.error?.message ? error.error.message : ""}`
```

Comment (Tbaut) [reply packages/files-ui/src/Contexts/FilesContext.tsx]: We can leave the last one at least
```suggestion
      let errorMessage = `${t`An error occurred: `} ${typeof(error) === "string" ? error : error?.error?.message || ""}`
```

Comment (RyRy79261) [reply packages/files-ui/src/Contexts/FilesContext.tsx]: Oh lord 😂

File: packages/files-ui/src/locales/de/messages.po [hunk 3]
```diff
@@ -20,7 +20,7 @@ msgid "A better sharing experience is coming soon."
 msgstr "Ein besseres Teilen-Erlebnis ist in Vorbereitung."
 
 msgid "A file with the same name already exists"
-msgstr "Es existiert bereits eine Datei mit demselben Namen"
+msgstr ""
```
Comment (Tbaut) [inline packages/files-ui/src/locales/de/messages.po:23]:
```suggestion
msgstr ""Es existiert bereits eine Datei mit demselben Namen""
```

File: packages/files-ui/src/locales/de/messages.po [hunk 4]
```diff
@@ -47,7 +47,7 @@ msgid "Adding you to the shared folder..."
 msgstr ""
 
 msgid "An error occurred:"
-msgstr "Es ist ein Fehler aufgetreten:"
+msgstr ""
```
Comment (Tbaut) [inline packages/files-ui/src/locales/de/messages.po:50]:
```suggestion
msgstr "Es ist ein Fehler aufgetreten:"
```

File: packages/files-ui/src/locales/de/messages.po [hunk 5]
```diff
@@ -692,7 +692,7 @@ msgid "Something went wrong!"
 msgstr "Etwas ist schief gelaufen!"
 
 msgid "Something went wrong. We couldn't upload your file"
-msgstr "Es ist etwas schief gelaufen. Wir konnten Ihre Datei nicht hochladen"
+msgstr ""
```
Comment (Tbaut) [inline packages/files-ui/src/locales/de/messages.po:695]:
```suggestion
msgstr "Es ist etwas schief gelaufen. Wir konnten Ihre Datei nicht hochladen"
```

File: packages/files-ui/src/locales/de/messages.po [hunk 6]
```diff
@@ -908,10 +908,10 @@ msgid "with"
 msgstr ""
 
 msgid "{0, plural, one {Downloading {1} file} other {Downloading {2} files}}"
-msgstr "{0, plural, one {{1} Datei wird heruntergeladen} other {{2} Dateien werden heruntergeladen}}"
+msgstr ""
```
Comment (Tbaut) [inline packages/files-ui/src/locales/de/messages.po:911]:
```suggestion
 msgstr "{0, plural, one {{1} Datei wird heruntergeladen} other {{2} Dateien werden heruntergeladen}}"
```

File: packages/files-ui/src/locales/de/messages.po [hunk 7]
```diff
@@ -908,10 +908,10 @@ msgid "with"
...
 
 msgid "{0, plural, one {Encrypting and uploading {1} file} other {Encrypting and uploading {2} files}}"
-msgstr "{0, plural, one {{1} Datei wird verschlüsselt und hochgeladen} other {{2} Dateien werden verschlüsselt und hochgeladen}}"
+msgstr ""
```
Comment (Tbaut) [inline packages/files-ui/src/locales/de/messages.po:914]:
```suggestion
 msgstr "{0, plural, one {{1} Datei wird verschlüsselt und hochgeladen} other {{2} Dateien werden verschlüsselt und hochgeladen}}"
```

File: packages/files-ui/src/locales/es/messages.po [hunk 8]
```diff
@@ -21,7 +21,7 @@ msgid "A better sharing experience is coming soon."
 msgstr ""
 
 msgid "A file with the same name already exists"
-msgstr "Ya existe un archivo con el mismo nombre"
+msgstr ""
```
Comment (Tbaut) [inline packages/files-ui/src/locales/es/messages.po:24]:
```suggestion
 msgstr "Ya existe un archivo con el mismo nombre"
```

File: packages/files-ui/src/locales/es/messages.po [hunk 9]
```diff
@@ -696,7 +696,7 @@ msgid "Something went wrong!"
 msgstr ""
 
 msgid "Something went wrong. We couldn't upload your file"
-msgstr "Algo salió mal. No pudimos subir tu archivo"
+msgstr ""
```
Comment (Tbaut) [inline packages/files-ui/src/locales/es/messages.po:699]:
```suggestion
 msgstr "Algo salió mal. No pudimos subir tu archivo"
```

File: packages/files-ui/src/locales/fr/messages.po [hunk 10]
```diff
@@ -22,7 +21,7 @@ msgid "A better sharing experience is coming soon."
 msgstr "Une meilleure expérience de partage sera bientôt disponible."
 
 msgid "A file with the same name already exists"
-msgstr "Un fichier avec ce nom existe déjà"
+msgstr ""
```
Comment (Tbaut) [inline packages/files-ui/src/locales/fr/messages.po:24]:
```suggestion
 msgstr "Un fichier avec ce nom existe déjà"
```

File: packages/files-ui/src/locales/fr/messages.po [hunk 11]
```diff
@@ -49,7 +48,7 @@ msgid "Adding you to the shared folder..."
 msgstr "Je vous ajoute au dossier partagé…"
 
 msgid "An error occurred:"
-msgstr "Une erreur s'est produite :"
+msgstr ""
```
Comment (Tbaut) [inline packages/files-ui/src/locales/fr/messages.po:51]:
```suggestion
 msgstr "Une erreur s'est produite :"
```

File: packages/files-ui/src/locales/fr/messages.po [hunk 12]
```diff
@@ -151,7 +150,7 @@ msgid "Copy to"
 msgstr "Copier vers"
 
 msgid "Copying files"
-msgstr "Copie de fichiers"
+msgstr ""
```
Comment (Tbaut) [inline packages/files-ui/src/locales/fr/messages.po:153]:
```suggestion
 msgstr "Copie de fichiers"
```

File: packages/files-ui/src/locales/fr/messages.po [hunk 13]
```diff
@@ -220,7 +219,7 @@ msgid "Download"
 msgstr "Télécharger"
 
 msgid "Download Complete"
-msgstr "Téléchargement terminé"
+msgstr ""
```
Comment (Tbaut) [inline packages/files-ui/src/locales/fr/messages.po:222]:
```suggestion
 msgstr "Téléchargement terminé"
```

**Review by Tbaut — CHANGES_REQUESTED** [review]
It's deleting a bunch of translation. I guess it's simpler to start from scratch :D

**Comment by Tbaut** [comment]
done in https://github.com/ChainSafe/ui-monorepo/pull/1717

**Pull request closed without merging**